### PR TITLE
Chore/sentry fixes 211024

### PR DESCRIPTION
**Pull request by joshenlim**
Resolves the following Sentry issues:
- undefined is not an object (evaluating 'm[0].slug') (on /authorize page)
  - https://supabase.sentry.io/issues/5808689343
  - https://supabase.sentry.io/issues/5562861576
- Cannot read properties of undefined (reading 'index') (on Storage Explorer)
  - https://supabase.sentry.io/issues/5998687405 
- (intermediate value)(intermediate value)(intermediate value).startsWith is not a function (in Users.utils.tsx)
  - https://supabase.sentry.io/issues/6002095756
- R.trim is not a function (in TextEditor.tsx)
  - https://supabase.sentry.io/issues/5414129652
  - https://supabase.sentry.io/issues/5999591797
- Add whitelist error for sign up mutation
  - https://supabase.sentry.io/issues/5562952529

### CODE REVIEW

**Review by alaister — APPROVED** [review]
LGTM!

- Tested on preview

File: apps/studio/components/to-be-cleaned/Storage/StorageExplorer/ColumnContextMenu.tsx [hunk 1]
```diff
@@ -60,15 +60,13 @@ const ColumnContextMenu = ({ id = '' }: ColumnContextMenuProps) => {
...
-        <>
-          <Item onClick={({ props }) => onSelectCreateFolder(props.index)}>
-            <FolderPlus size="14" strokeWidth={1} />
-            <span className="ml-2 text-xs">New folder</span>
-          </Item>
-          <Separator />
-        </>
-      )}
+      {canUpdateFiles && [
+        <Item key="create-folder" onClick={({ props }) => onSelectCreateFolder(props.index)}>
```
Comment (ivasilov) [inline apps/studio/components/to-be-cleaned/Storage/StorageExplorer/ColumnContextMenu.tsx:64]: Does this really fixes https://supabase.sentry.io/issues/5998687405?project=5459134?

Comment (joshenlim) [reply apps/studio/components/to-be-cleaned/Storage/StorageExplorer/ColumnContextMenu.tsx:64]: yup! this is a slight oddity to conditionally render react-contextify `Items` where they must be rendered as an array and not within a fragment - i'm not sure why this is the case though, but we did do something similar [here](https://github.com/supabase/supabase/blob/master/apps/studio/components/to-be-cleaned/Storage/StorageExplorer/ItemContextMenu.tsx#L86) as well

Comment (joshenlim) [reply apps/studio/components/to-be-cleaned/Storage/StorageExplorer/ColumnContextMenu.tsx:64]: you can verify this bug if you check local `master` atm - right click in the storage explorer on a column and select new folder, a UI error will show up

Comment (ivasilov) [reply apps/studio/components/to-be-cleaned/Storage/StorageExplorer/ColumnContextMenu.tsx:64]: Yup, you're right. Checked on prod and preview. LGTM.

**Review by ivasilov — APPROVED** [review]

**Pull request merged**